### ISSUE TITLE
Composer Hosted - New package version not in package JSON ...

### ISSUE THREAD

**Issue by jcolfej** [sonatype-nexus-community/nexus-repository-composer#31]
I try to add new versions of my package on a hosted composer repo.
New versions don't appear on `https://my.nexus/repository/composer_hosted/p/vendor/package.json` ...
I only have my first package version.

Currently, I push a 1.0.0 version and few minutes after, I push 1.0.1, but json say :

```
{
  "packages": {
    "vendor/package": {
      "1.0.0": {
        "name": "vendor/package",
        "version": "1.0.0",
        "dist": {
          "url": "https://my.nexus/repository/composer_hosted/vendor/package/1.0.0/vendor/package-1.0.0.zip",
          "type": "zip",
          "reference": "xXxXxXxXxXxXxXxXxXxXxXxXxXxXxXxXxXxXxXx",
          "shasum": "xXxXxXxXxXxXxXxXxXxXxXxXxXxXxXxXxXxXxXx"
        },
        "time": "2018-10-01T12:39:52+00:00",
        "uid": 3251794098,
        "autoload": {
          "psr-4": {
            "Vendor\\Package\\": "src/"
          }
        },
        "authors": [
          {
            "name": "Me",
            "email": "me@me.me"
          }
        ],
        "description": "My package ..."
      }
    }
  }
}
```

Is anyone have same problem ?

**Comment by jcolfej** [comment]
After some tests, this problem is only in case if you use S3 as blobstore.

**Comment by DarthHater** [comment]
That is interesting. @fjmilens3 or @ataylor284 might be interested.

**Comment by bcastagna** [comment]
I have the exact same problem and I think I dont use S3 as blobstore, just "default" blob storage type.
I have posted on this [issue ](https://github.com/sonatype-nexus-community/nexus-repository-composer/issues/21) few days ago because I thought the "Rebuild All Composer MetaData task" could solve this but if this task intend to do the same thing as the "Rebuild Index" button in the repo's admin panel, then it wont work.
@fjmilens3 @TheBay0r Could you guys tell us what could cause this ?

**Comment by bcastagna** [comment]
Any news on this issue ? Please could someone point us why it does not update the JSON ? It's really a pity because it is the last small issue that prevents me from using this full-scale....
@DarthHater @ataylor284 
@fjmilens3 Are you okay man ? 4 months without ANY commits makes me worry.

**Comment by codercms** [comment]
I ran into the same problem. But I found solution, in settings of composer-hosted repository you should allow redeploy.
![image](https://user-images.githubusercontent.com/4526911/53393761-47034480-39ae-11e9-93ba-fdd2f2c30599.png)

**Comment by DarthHater** [comment]
@bcastagna sadly @fjmilens3 has moved on to greener pastures from Sonatype, which doesn't mean he won't pop back in here, but it would be out of the kindness of his big heart (he has one for sure). I've been fairly busy helping get another person scaled up on working on this type of stuff, and we are working through a backlog on some other issues right now. I'll try and get either @bhamail or myself to pop in here some time soon!

**Comment by zhwei** [comment]
hi guys, i have the same problem, is there any update to tetram's pr ?

**Comment by davidferlay** [comment]
Encountered 

> I ran into the same problem. But I found solution, in settings of composer-hosted repository you should allow redeploy.
> ![image](https://user-images.githubusercontent.com/4526911/53393761-47034480-39ae-11e9-93ba-fdd2f2c30599.png)

+++ 

This is quite a critical issue imho
"Allow redeploy" is a very dangerous settings because it allows packages to be overwritten if an update is made but the version is not increased : it will break all projects where this package version was already installed (and is present in composer.lock) > `composer install` will always fail because package sha changed.

**Comment by bhamail** [comment]
I merged PR #44 from @tetram. Please give it a try and report any issues.
